multi-int-store/src/main/java/com/connexta/multiintstore/config/S3StorageConfiguration.java

```diff
@@ -33,42 +33,44 @@
 public class S3StorageConfiguration {
 
   @Bean
-  @Profile("s3Production")
-  public AmazonS3 amazonS3(
-      @Value("${aws.s3.endpointUrl}") @NotBlank final String endpoint,
-      @Value("${aws.s3.region}") @NotBlank final String region,
-      @Value("${aws.s3.secret.file}") @NotBlank final String secretKeyFile,
-      @Value("${aws.s3.access.file}") @NotBlank final String accessKeyFile)
-      throws IOException {
-    final AwsClientBuilder.EndpointConfiguration endpointConfiguration =
-        new AwsClientBuilder.EndpointConfiguration(endpoint, region);
-    final BasicAWSCredentials credentials =
-        new BasicAWSCredentials(
-            FileUtils.readFileToString(new File(accessKeyFile), StandardCharsets.UTF_8),
-            FileUtils.readFileToString(new File(secretKeyFile), StandardCharsets.UTF_8));
-    final AmazonS3 s3Client =
-        AmazonS3ClientBuilder.standard()
-            .withCredentials(new AWSStaticCredentialsProvider(credentials))
-            .withEndpointConfiguration(endpointConfiguration)
-            .build();
-    log.info("S3 Client has been initialized.");
-    log.info("Region: {}", region);
-    log.info("Endpoint: {}", endpoint);
-    return s3Client;
+  public AmazonS3 amazonS3(@NotNull final AmazonS3Configuration configuration) {
+    return AmazonS3ClientBuilder.standard()
+        .withEndpointConfiguration(
+            new EndpointConfiguration(configuration.getEndpoint(), configuration.getRegion()))
+        .withCredentials(
+            new AWSStaticCredentialsProvider(
+                new BasicAWSCredentials(
+                    configuration.getAccessKey(), configuration.getSecretKey())))
+        .enablePathStyleAccess()
+        .build();
+  }
+
+  // TODO does this need to be a separate bean?
```

Comment:
No, we should be able to initialize it in the StorageAdaptor bean since all Spring beans are singletons by default.

Reply:
Let's address this in a future PR. I'll leave the TODO comment in there.